spec/support/shared_examples.rb

```diff
@@ -0,0 +1,35 @@
+shared_examples 'a patcher' do
+  let(:patcher) { described_class.new(patch_location) }
+  let(:patch_location) { nil }
+
+  describe '#mutate_schema' do
+    it 'implements mutate schema' do
+      expect(patcher).to respond_to(:mutate_schema)
```

Comment:
I'm not sure this does what we're wanting. This would always return `true`, because we have `mutate_schema` in the base classes (even though all they do is `raise NotImplementedError`). Probably what we mean here is `expect { patcher.mutate_schema }.not_to raise_error(NotImplementedError)` if we want what we're testing to reflect the `it` string.